librz/analysis/fcn.c

```diff
@@ -2308,15 +2308,33 @@ RZ_API RZ_OWN RzPVector *rz_analysis_function_args(RzAnalysis *a, RzAnalysisFunc
 	if (!a || !fcn) {
 		return NULL;
 	}
-	RzPVector *args = rz_pvector_new(NULL);
+	RzPVector *tmp = rz_pvector_new(NULL);
+	RzAnalysisVar *var;
 	void **it;
+	int rarg_idx = 0;
+	//Resort the pvector to order reg_arg - stack_arg
 	rz_pvector_foreach (&fcn->vars, it) {
-		RzAnalysisVar *var = *it;
+		var = *it;
+		if (var->kind == RZ_ANALYSIS_VAR_KIND_REG) {
+			rz_pvector_insert(tmp, rarg_idx++, var);
+		} else {
+			rz_pvector_push(tmp, var);
+		}
+	}
```

Comment:
isn't better to push everything and call `rz_pvector_sort`?

Reply:
The `RzAnalysisVarKind` enum is declared as below:
``` C
typedef enum {
	RZ_ANALYSIS_VAR_KIND_REG = 'r', //114
	RZ_ANALYSIS_VAR_KIND_BPV = 'b', //98
	RZ_ANALYSIS_VAR_KIND_SPV = 's' //115
} RzAnalysisVarKind;
```
So if we call `rz_pvector_sort`, it will be the order of "BP-REG-SP or reverse".

Reply:
Yes, for now we should treat them differently.